pkg/syncutil/aggregator/aggregator.go

```diff
@@ -46,7 +46,7 @@ func (b *GVKAgreggator) IsPresent(gvk schema.GroupVersionKind) bool {
 	return found
 }
 
-// Remove deletes the any associations that Key k has in the GVKAggregator.
+// Remove deletes any associations that Key k has in the GVKAggregator.
 // For any GVK in the association k --> [GVKs], we also delete any associations
 // between the GVK and the Key k stored in the reverse map.
 func (b *GVKAgreggator) Remove(k Key) error {
```

Comment:
How can Remove error?

Reply:
from the pruning of reverse edges. IIRC we discussed the trade off back when we wrote the aggregator. Happy to chat some more but would love to focus on the cache manager here.